docs/tuning/README.md

```diff
@@ -122,22 +89,52 @@ The dataset should follow a specific format. For example:
+
+
+If your dataset is not in one of these formats, it will be passed directly to the training library ([SFTTrainer](https://huggingface.co/docs/trl/en/sft_trainer)) without any preprocessing. This may result in undefined behavior if the dataset does not align with the trainer's expected input structure. To ensure proper functionality, you may need to preprocess the dataset to match one of the supported formats. For more details, please refer to this [documentation](https://huggingface.co/docs/trl/v0.9.4/sft_trainer#dataset-format-support).
 
 
-## Dataset Format Support
+Note: if you build a container image for the input dataset, please copy the dataset to the **`/data`** directory inside the container.
 
-The SFTTrainer supports popular dataset formats, allowing direct passage of the dataset without pre-processing. Supported formats include conversational and instruction formats. If your dataset is not in one of these formats, it will be passed directly to the SFTTrainer without any preprocessing. This may result in undefined behavior if the dataset does not align with the trainer's expected input structure.
+# Tuning Job
+Kaito uses the Kubernetes **batchv1.job** workload to manage the tuning Pod. When a tuning workspace custom resource is created, Kaito controller will create a job with the same name of the workspace in the same namespace. To streamline the tuning workflow, Kaito adds two containers in addition to the main container that runs the tuning process. The pod structure is illustrated in Figure 1.
+<div align="left">
+  <img src="../img/kaito-fine-tuning.png" width=40% title="Kaito fine tuning" alt="Kaito fine tuning">
+</div>
+Figure 1. Kaito tuning pod structure.
 
-To ensure proper function, you may need to preprocess the dataset to match one of the supported formats.
+An initcontainer `data-downloader` is introduced to download the training input dataset from the URLs specified in the tuning spec. In case an image is specified in the input, the `data-downloader` container simply uses the specified image as the container image. This initcontainer ensures the training data is available locally before the training process starts.
+
+To support automatically pushing the tuning results to a container registry, Kaito adds a sidecar container which has `docker` installed. This container runs a script which periodically checks the training progress. Once the training is done, indicated by a sentinel file created by the training process, the script will build a container image containing the training results and push the image to the specified container registry.
+
+The main container uses one of the supported model images. The image entry launches the [fine\_tuning.py](https://github.com/Azure/kaito/blob/main/presets/tuning/text-generation/fine_tuning.py) script.
+
+All three containers use shared local volumes (by mounting same `EmptyDir` volumes). Hence file copies between containers are avoided.
```

Comment:
All three containers use shared local volumes (by mounting the same `EmptyDir` volumes), hence file copies between containers are avoided.